src/components/Pagination.ts

```diff
@@ -0,0 +1,361 @@
+                    }
+                    if (breakViewAdded) {
+                        breakViewAdded = false;
+                    }else {
+                        pageItems.push(this.getPageNumberView(page));
+                    }
+                    continue;
+                }
+
+                if (page === 2) {
+                    pageItems.push(createElement(BreakView, {}));
+                    breakViewAdded = true;
+                    continue;
+                }
+
+                if (this.state.selectedPageNumber <= leftSide && page === maxPageButtons) {
+                    pageItems.push(this.getPageNumberView(page));
+                    continue;
+                }
+
+                if (page === pageCount - 1) {
+                    pageItems.push(createElement(BreakView, {}));
+                }
+            }
+        }
+
+        return pageItems;
+    }
+
+    private getPageNumberView(pageNumber: number) {
+        return createElement(PageNumberView, {
+            key: `key${pageNumber}`,
+            onClick: () => this.handleSelectedPage(pageNumber),
+            page: pageNumber,
+            selected: this.state.selectedPageNumber === pageNumber
+        });
+    }
+
+    private handleSelectedPage(selectedPageNumber: number) {
+        const currentOffset = (selectedPageNumber - 1) * this.props.offset;
+
+        if (this.state.selectedPageNumber === selectedPageNumber) {
+            return;
+        }
+
+        this.setState({
+            currentOffset,
+            nextIsDisabled: (currentOffset + this.props.offset) === this.props.listViewSize,
+            previousIsDisabled: currentOffset === 0,
+            selectedPageNumber
+        });
+
+        this.props.onClickAction(currentOffset);
+    }
+
+    private getMessageStatus(message?: string): string {
+        const currentOffset = this.state.currentOffset;
+        const { listViewSize, offset } = this.props;
+        let fromValue = currentOffset + 1;
+        let toValue = 0;
+
+        if (listViewSize === 0) {
+            fromValue = 0;
+        } else if (listViewSize < offset || (currentOffset + offset) > listViewSize) {
+            toValue = listViewSize;
+        } else {
+            toValue = currentOffset + offset;
+        }
+
+        if (message) {
+            return message
+                .replace("{firstItem}", fromValue.toString())
+                .replace("{lastItem}", toValue.toString())
+                .replace("{totalItems}", listViewSize.toString())
+                .replace("{currentPageNumber}", this.state.selectedPageNumber.toString())
+                .replace("{totalPages}",
+                    (
+                        offset && offset !== 0
+                            ? Math.ceil(listViewSize / offset)
+                            : listViewSize
+                    ).toString());
+        }
+
+        return this.props.getMessageStatus(fromValue, toValue, listViewSize);
+    }
+}
```

Comment:
File is to long, should be split.

Reply:
Given that delivery and tasks at hand,  this will affect delivery